community/values/src/main/java/org/neo4j/values/storable/DurationValue.java

```diff
@@ -0,0 +1,805 @@
+    {
+        int sign = "-".equals( matcher.group( "sign" ) ) ? -1 : 1;
+        String y = matcher.group( "years" );
+        String m = matcher.group( "months" );
+        String w = matcher.group( "weeks" );
+        String d = matcher.group( "days" );
+        String t = matcher.group( "T" );
+        if ( (y == null && m == null && w == null && d == null && t == null) || "T".equalsIgnoreCase( t ) )
+        {
+            return null;
+        }
+        long months = optLong( y ) * 12 + optLong( m );
+        long days = optLong( w ) * 7 + optLong( d );
+        return parseDuration( sign, months, days, matcher, false, "hours", "minutes", "seconds", "subseconds" );
+    }
+
+    private static DurationValue parseDateDuration( String year, Matcher matcher, boolean time )
+    {
+        int sign = "-".equals( matcher.group( "sign" ) ) ? -1 : 1;
+        long months = 0;
+        long days = 0;
+        if ( year != null )
+        {
+            String month = matcher.group( "longMonth" );
+            String day;
+            if ( month == null )
+            {
+                month = matcher.group( "shortMonth" );
+                day = matcher.group( "shortDay" );
+            }
+            else
+            {
+                day = matcher.group( "longDay" );
+            }
+            months = parseLong( month );
+            if ( months > 12 )
+            {
+                throw new IllegalArgumentException( "months is out of range: " + month );
+            }
+            months += parseLong( year ) * 12;
+            days = parseLong( day );
+            if ( days > 31 )
+            {
+                throw new IllegalArgumentException( "days is out of range: " + day );
+            }
+        }
+        if ( time )
+        {
+            if ( matcher.group( "longHour" ) != null )
+            {
+                return parseDuration(
+                        sign, months, days, matcher, true, "longHour", "longMinute", "longSecond", "longSub" );
+            }
+            else
+            {
+                return parseDuration(
+                        sign, months, days, matcher, true, "shortHour", "shortMinute", "shortSecond", "shortSub" );
+            }
+        }
+        else
+        {
+            return duration( sign * months, sign * days, 0, 0 );
+        }
+    }
+
+    private static DurationValue parseDuration(
+            int sign, long months, long days, Matcher matcher, boolean strict,
+            String hour, String min, String sec, String sub )
+    {
+        long h = optLong( matcher.group( hour ) );
+        long m = optLong( matcher.group( min ) );
+        String s = matcher.group( sec );
+        String n = matcher.group( sub );
+        long seconds = optLong( s );
+        if ( strict )
+        {
+            if ( h > 24 )
+            {
+                throw new IllegalArgumentException( "hours out of range: " + h );
+            }
+            if ( m > 60 )
```

Comment:
I could only find the  valid ranges for minute, second, etc. for the `Time` types. If the caps are at 59 for minutes and 60 for seconds _there_, why are they both at 60 for durations?

Reply:
good point, should have been `>=`

Reply:
Classic kind of thing that should be covered by a test case :)